syft/pkg/cataloger/nix/cataloger_test.go

```diff
@@ -0,0 +1,55 @@
+package nix
+
+import (
+	"testing"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/internal/pkgtest"
+	"github.com/anchore/syft/syft/source"
+)
+
+func TestCataloger_Catalog(t *testing.T) {
+
+	tests := []struct {
+		fixture  string
+		wantPkgs []pkg.Package
+		wantRel  []artifact.Relationship
+	}{
+		{
+			fixture: "test-fixtures/fixture-1",
+			wantPkgs: []pkg.Package{
+				{
+					Name:         "glibc",
+					Version:      "2.34-210",
```

Comment:
According to how I interpret [@flokli's comment](https://github.com/anchore/syft/pull/1696#discussion_r1148320063), the version here should be `2.34`, not including the `-210` (maybe it's the build number or something?)

Reply:
That particular comment is referring to the `-bin` suffix, which refers to the specific "output" called `bin`. The version for glibc does appear to have the `-###` suffix (which is different than the output):
<img width="510" alt="Screenshot 2023-04-03 at 1 37 57 PM" src="https://user-images.githubusercontent.com/590471/229585764-c9f768bc-339b-4b39-a370-e85a58d18914.png">